zapcore/encoder.go

```diff
@@ -22,6 +22,31 @@ package zapcore
 
 import "io"
 
+// ObjectEncoder is a strongly-typed, encoding-agnostic interface for adding a
+// map[string]interface{} to the logging context. Like maps, ObjectEncoders
+// aren't safe for concurrent use (though typical use shouldn't require locks).
+type ObjectEncoder interface {
+	AddBool(key string, value bool)
+	AddFloat64(key string, value float64)
+	AddInt64(key string, value int64)
+	AddUint64(key string, value uint64)
+	AddObject(key string, marshaler ObjectMarshaler) error
+	AddArray(key string, marshaler ArrayMarshaler) error
+	// AddReflected uses reflection to serialize arbitrary objects, so it's slow
+	// and allocation-heavy.
+	AddReflected(key string, value interface{}) error
+	AddString(key, value string)
+}
```

Comment:
Moved most of this  here from `object_encoder.go`, since it seemed silly to have two interface-only files for encoders.